packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -1,14 +1,301 @@
+    this.distributionId = distribution.ref;
+  }
+
+  /**
+   * Adds a new behavior to this distribution for the given pathPattern.
+   *
+   * @param pathPattern the path pattern (e.g., 'images/*') that specifies which requests to apply the behavior to.
+   * @param behaviorOptions the options for the behavior at this path.
+   */
+  public addBehavior(pathPattern: string, behaviorOptions: BehaviorOptions) {
+    if (pathPattern === '*') {
+      throw new Error('Only the default behavior can have a path pattern of \'*\'');
+    }
+    this.additionalBehaviors.push(new CacheBehavior({ pathPattern, ...behaviorOptions }));
+  }
+
+  private renderOrigins(): CfnDistribution.OriginProperty[] {
+    const origins = new Set<Origin>();
+    origins.add(this.defaultBehavior.origin);
+    this.additionalBehaviors.forEach(behavior => origins.add(behavior.origin));
+
+    const renderedOrigins: CfnDistribution.OriginProperty[] = [];
+    origins.forEach(origin => renderedOrigins.push(origin._renderOrigin()));
+    return renderedOrigins;
+  }
+
+  private renderCacheBehaviors(): CfnDistribution.CacheBehaviorProperty[] | undefined {
+    if (this.additionalBehaviors.length === 0) { return undefined; }
+    return this.additionalBehaviors.map(behavior => behavior._renderBehavior());
+  }
+
+  private renderCustomErrorResponses(): CfnDistribution.CustomErrorResponseProperty[] | undefined {
+    if (this.errorConfigurations.length === 0) { return undefined; }
+    function validateCustomErrorResponse(errorResponse: CfnDistribution.CustomErrorResponseProperty) {
+      if (errorResponse.responsePagePath && !errorResponse.responseCode) {
+        throw new Error('\'responseCode\' must be provided if \'responsePagePath\' is defined');
+      }
+      if (!errorResponse.responseCode && !errorResponse.errorCachingMinTtl) {
+        throw new Error('A custom error response without either a \'responseCode\' or \'errorCachingMinTtl\' is not valid.');
+      }
+    }
+    this.errorConfigurations.forEach(e => validateCustomErrorResponse(e));
+    return this.errorConfigurations;
+  }
+
+}
+
+/**
+ * The price class determines how many edge locations CloudFront will use for your distribution.
+ */
+export enum PriceClass {
+  PRICE_CLASS_100 = 'PriceClass_100',
+  PRICE_CLASS_200 = 'PriceClass_200',
+  PRICE_CLASS_ALL = 'PriceClass_All'
+}
+
+/**
+ * How HTTPs should be handled with your distribution.
+ */
+export enum ViewerProtocolPolicy {
+  HTTPS_ONLY = 'https-only',
+  REDIRECT_TO_HTTPS = 'redirect-to-https',
+  ALLOW_ALL = 'allow-all'
+}
+
+/**
+ * Defines what protocols CloudFront will use to connect to an origin.
+ */
+export enum OriginProtocolPolicy {
+  HTTP_ONLY = 'http-only',
+  MATCH_VIEWER = 'match-viewer',
+  HTTPS_ONLY = 'https-only',
+}
+
+/**
+ * The HTTP methods that the Behavior will accept requests on.
+ */
+export class AllowedMethods {
+  /** HEAD and GET */
+  public static readonly ALLOW_GET_HEAD = new AllowedMethods(['GET', 'HEAD']);
```

Comment:
I think the `ALLOW_` prefix can be dropped. The access is going to always be `AllowedMethods.ALL`...

Reply:
:D
I have three options here:
1. GET_HEAD - Aligns with the ordering always used in the documentation; puts the GET (more important IMO) in front. Isn't currently allowed by jsii.
2. HEAD_GET - Flipped to get around jsii issue.
3. ALLOW_GET_HEAD - Solves jsii, also looks nice for those types of Java folk who statically import enums.

The jsii issue is that GET_ isn't allowed as a prefix (to prevent over-shadowing getters in Java). The rule _should_ exclude static readonly and/or ALL_CAPS versions, but currently doesn't. I can submit the fix to jsii, but then am going to have to wait for that to propagate everywhere, etc. In talking with @RomainMuller, he suggested the ALLOW_* versions.

Thoughts?

Reply:
allow_* sounds reasonable. the world is messy